packages/server/src/adapters/lambda/index.ts

```diff
@@ -0,0 +1,156 @@
+    headers: event.headers,
+    body: event.body,
+  };
+}
+
+function getHTTPMethod(event: APIGatewayEvent) {
+  if (isPayloadV1(event)) {
+    return event.httpMethod;
+  } else if (isPayloadV2(event)) {
+    return event.requestContext.http.method;
+  } else {
+    throw new TRPCError({
+      code: 'INTERNAL_SERVER_ERROR',
+      message: UNKNOWN_PAYLOAD_FORMAT_VERSION_ERROR_MESSAGE,
+    });
+  }
+}
+function getPath(event: APIGatewayEvent) {
+  if (isPayloadV1(event)) {
+    return event.path.slice(1);
+  } else if (isPayloadV2(event)) {
+    return event.rawPath.slice(1);
+  } else {
+    throw new TRPCError({
+      code: 'INTERNAL_SERVER_ERROR',
+      message: UNKNOWN_PAYLOAD_FORMAT_VERSION_ERROR_MESSAGE,
+    });
+  }
+}
+function transformHeaders(headers: HTTPHeaders): APIGatewayResult['headers'] {
+  const obj: APIGatewayResult['headers'] = {};
+
+  for (const [key, value] of Object.entries(headers)) {
+    if (typeof value === 'undefined') {
+      continue;
+    }
+    obj[key] = Array.isArray(value) ? value.join(',') : value;
+  }
+  return obj;
+}
+function tRPCOutputToAPIGatewayOutput<
+  TEvent extends APIGatewayEvent,
+  TResult extends APIGatewayResult,
+>(event: TEvent, response: HTTPResponse): TResult {
+  if (isPayloadV1(event)) {
+    const resp: APIGatewayProxyResult = {
+      statusCode: response.status,
+      body: response.body ?? '',
+      headers: transformHeaders(response.headers ?? {}),
+    };
+    return resp as TResult;
+  } else if (isPayloadV2(event)) {
+    const resp: APIGatewayProxyStructuredResultV2 = {
+      statusCode: response.status,
+      body: response.body ?? undefined,
+      headers: transformHeaders(response.headers ?? {}),
+    };
+    return resp as TResult;
+  } else {
+    throw new TRPCError({
+      code: 'INTERNAL_SERVER_ERROR',
+      message: UNKNOWN_PAYLOAD_FORMAT_VERSION_ERROR_MESSAGE,
+    });
+  }
+}
+
+/** Will check the createContext of the TRouter and get the parameter of event.
+ * @internal
+ **/
+type inferAPIGWEvent<
+  TRouter extends AnyRouter,
+  TEvent extends APIGatewayEvent,
+> = AWSLambdaOptions<TRouter, TEvent>['createContext'] extends NonNullable<
+  AWSLambdaOptions<TRouter, TEvent>['createContext']
+>
+  ? Parameters<AWSLambdaOptions<TRouter, TEvent>['createContext']>[0]['event']
+  : APIGatewayEvent;
+
+/** 1:1 mapping of v1 or v2 input events, deduces which is which.
+ * @internal
+ **/
+type inferAPIGWReturn<T> = T extends APIGatewayProxyEvent
+  ? APIGatewayProxyResult
+  : T extends APIGatewayProxyEventV2
+  ? APIGatewayProxyStructuredResultV2
+  : never;
+export function lambdaRequestHandler<
+  TRouter extends AnyRouter,
+  TEvent extends inferAPIGWEvent<TRouter, TEvent>,
+  TResult extends inferAPIGWReturn<TEvent>,
+>(
+  opts: AWSLambdaOptions<TRouter, TEvent>,
+): (event: TEvent, context: APIGWContext) => Promise<TResult> {
+  return async (event, context) => {
+    const req = lambdaEventToHTTPRequest(event);
+    const path = getPath(event);
```

Comment:
It looks like if these fail, we don't get a response sent to the client, right?

Reply:
No, that's correct. But what types of failures would you expect?

Reply:
I’m not 100% sure but I believe Aws maps rejected promises to some kind of 5xx response